onnxruntime/core/providers/webnn/builders/helper.h

```diff
@@ -54,6 +54,10 @@ std::string GetShapeString(std::vector<T>& shape) {
   return shape_info.str();
 }
 
+inline std::string GetTensorName(const ConstPointerContainer<std::vector<NodeArg *>>& input_defs, const size_t index) {
```

Comment:
Can we avoid the std::string allocations each call by using C++17 std::string_view instead?

nit extra space `std::vector<NodeArg*>`

Reply:
The `Contains(initializers, name)` complains the `std::string_view`, as the `initializers` is typeof `InitializedTensorSet` which uses `std::string` as key.
```C++
using InitializedTensorSet = std::unordered_map<std::string, const ONNX_NAMESPACE::TensorProto*>;
```